limitador-server/examples/envoy.yaml

```diff
@@ -29,7 +29,7 @@ static_resources:
                     - stage: 0
                       actions:
                         - {request_headers: {header_name: "userid", descriptor_key: "user_id"}}
-                        - {request_headers: {header_name: ":method", descriptor_key: "req.method"}}
+                        - { request_headers: { header_name: ":method", descriptor_key: "req_method" } }
```

Comment:
What if, as a user, I want to send descriptor entry key `req.method`? 

If _Limit_'s _Variable_ and _Condition_ are full CEL expressions, there should be a way to make reference to the descriptor entries. Maybe something like:

```
req.entry["req.method"] == 'POST'
```

wdyt?

Reply:
That's indeed another reason for the ["root" binding I'm arguing for](https://github.com/Kuadrant/limitador/pull/396#discussion_r1861201539)... 
so `req` doesn't mean anything... right now, we use the descriptor entries to infer some naming... We _could_ do some massaging when getting the values inject still too... e.g. treat the dot (`.`) as some special thing and create a map... e.g. when getting `req.method` as a descriptor key, it'd bind to a map `req` with an entry `method`, which would point to the value of the descriptor... but that's a never ending story... what about spaces (` `), question marks (`?`), colons (`:`), ... ? More so, what if there is an entry `limit` in one of the descriptors?

So I'd "just" embrace the fact that they can be any character and put them all in a map... but what? Each and every discreet descriptor, i.e. `descriptors[0]["req.method:foo?bar"]` ?

Reply:
Also, to be clear, the `.` navigation works transparently on `Map`:

```
descriptor[0].foo == descriptor[0]['foo']
```